File: tests/fixtures/adapter-parity/get-user.go
package service

import "fmt"

// User represents a user entity.
type User struct {
	ID    int    `json:"id"`
	Name  string `json:"name"`
	Email string `json:"email"`
}

// GetUser fetches a user by ID.
func GetUser(id string) (*User, error) {
	if id == "" {
		return nil, fmt.Errorf("get user: id is empty. Provide a valid user ID.")
	}
	return &User{ID: 1, Name: "Alice", Email: "alice@example.com"}, nil
}

// CreateUser creates a new user.
func CreateUser(name string, email string) (*User, error) {
	if name == "" {
		return nil, fmt.Errorf("create user: name is empty. Provide a name.")
	}
	return &User{ID: 2, Name: name, Email: email}, nil
}

File: tests/fixtures/adapter-parity/get-user.ts
// get-user.ts — User service module.

export interface User {
  id: number;
  name: string;
  email: string;
}

export async function getUser(id: string): Promise<User> {
  if (!id) {
    throw new Error("get user: id is empty. Provide a valid user ID.");
  }
  return { id: 1, name: "Alice", email: "alice@example.com" };
}

export async function createUser(name: string, email: string): Promise<User> {
  if (!name) {
    throw new Error("create user: name is empty. Provide a name.");
  }
  return { id: 2, name, email };
}
